students/Sally_Shen/lesson5/MailroonPart3.py

```diff
@@ -0,0 +1,142 @@
+
+'''choms'''
+
+import sys
+import os
+
+print("in mailroom")
+'''Writing first name is easier for testing.'''
+donors ={"Fred Gates": [100.90, 200, 50], "Jeff Bezos": [100, 50, 600], "Mark Zuckerberg": [200, 50, 500],
+          "Paul Allen": [300, 40, 200]}
+
+
+# donors = {{"first_name": "Fred", "last_name": "Gates"}: [[100.90, 200, 50], 265.98]}
+
+
+def thank_you():
+    response = input("Type your donor name, if don't know, type list to pick")
+    '''If the user types list show them a list of the donor names and re-prompt'''
+    if response == "list":
+        for key in donors.keys():
+            print(key)
+        response = input("Please type your donor name.")
+
+    if response not in donors.keys():
+        donors[response] = []
+
+    newDonation = input("how much to donate?")
+    validInput = False
+
+    while not validInput:
+        try:
+            donors[response].append(float(newDonation))
+            validInput = True
+        except ValueError:
+            newDonation = input("Please input numbers!")
+
+    email = '''
+    Dear {0},
+    Thank you for donating {1}!\n'''.format(response, newDonation)
+    return email
+
+
+def gen_stats(donor):
+    donations = donor[1]
+    total = sum(donations)
+    num = len(donations)
+    stats = (donor[0], total, num, total / num)
+    return
+
+
+def report():
+    amountToNames = {}
+    for name, amount in donors.items():
+        total = sum(amount)
+        if total in amountToNames.keys():
+            amountToNames[total].append(name)
+        else:
+            amountToNames[total] = [name]
+
+    sortedAmount = sorted(amountToNames, reverse=True)
+    output = "{:<20}{:<20}{:<15}{:<20}\n".format("Donor Name", "|Total Given", "|Num Gifts", "|Average Gift")
+    output += "-------------------------------------------------------------------\n"
+
+
+    '''  
+    for amount in sortedAmount:
+        nameList = amountToNames[amount]
+        for name in nameList:
+            numGifts = len(donors[name])
+            output += lineItem(name, amount, numGifts, amount / numGifts)
+            
+    '''
+    '''Change the code by using comprehensions'''
+
+    output += "".join(
+        lineItem(name, amount, len(donors[name]), amount / len(donors[name])) for amount in sortedAmount for name in amountToNames[amount])
+
+    print(output)
+
+
+
+def lineItem(name, total, numGifts, avg):
+    output = "{:<20}${:<20.2f}{:<15}${:<20.2f}\n".format(name, total, numGifts, avg)
+    return output
+
+
+def gen_thank_you_letters(a_dict):
+    dir = "thankyouLetters"
+    try:
+        os.mkdir(dir)
+    except FileExistsError:
+        pass
+    for name, amount in donors.items():
+        firstLastName = name.split()
+        fileName = firstLastName[0] + "_" + firstLastName[1] + ".txt"
+        filePath = dir + "\\" + fileName
```

Comment:
This ends up prepending dir + \\ into the filename. What you want to do is change the working directory to enter the filePath and then create the files.
For example:
os.chdir(path)

Reply:
Thank you, fixed in lesson 6 PR